### DISCUSSION TITLE
Strange error

### DISCUSSION THREAD

**Discussion by zxcqirara** [rwf2/Rocket#2607] [Questions] [answered]
I need to upload big file, set max size to 5GB, but on upload 1.7GB file it throws this
```
   >> Data guard `TempFile < '_ >` failed: Custom { kind: Other, error: hyper::Error(Body, Custom { kind: UnexpectedEof, error: IncompleteBody }) }.
   >> No 400 catcher registered. Using Rocket default.
```

**Comment by SergioBenitez** [comment]
How are you uploading the file? Is your client aborting the operation? Is the temporary file location `temp_dir` backed by memory or disk, and is there enough memory/disk space available?

**Reply by zxcqirara** [reply]
Client didn't abort an operation, temp dir was located on disk, there was enough space available. Error went when I used [data stream](https://rocket.rs/v0.5-rc/guide/requests/#streaming) instead of temp files

**Reply by zxcqirara** [reply]
I think it may happens due to request timeout, but not sure

**Reply by SergioBenitez** [reply, marked as answer]
According to the log, the client is closing the connection for some reason. In other words, it's the client, not Rocket. It would be prudent to understand why.

**Reply by zxcqirara** [reply]
I used reqwest, now I use ureq, the problem went, so, I think you are right, maybe not rocket trouble...

**Comment by zxcqirara** [comment]
Well, problem was with reqwest client, after change it to ureq problem problem disappeared
